crates/natrix/tests/actual_tests/derive_macros.rs

```diff
@@ -0,0 +1,221 @@
+use natrix::prelude::*;
+use wasm_bindgen_test::{wasm_bindgen_test, wasm_bindgen_test_configure};
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+// Test the Project derive macro with additional derives
+#[derive(Project)]
+#[project(Debug)]
+enum TestEnum<T> {
+    Value(T),
+    Text(String),
+    Empty,
+}
+
+#[derive(Project)]
+enum SimpleEnum {
+    A,
+    B,
+    C,
+}
+
+// Test Downgrade derive
+#[derive(Downgrade)]
+enum DowngradeTestEnum<T> {
+    Value(T),
+    Empty,
+}
+
+#[wasm_bindgen_test]
+fn project_derive_compiles() {
+    // Test that the derive macro generates valid code
+    // This is just a compilation test for now
+    let _test_enum: TestEnum<i32> = TestEnum::Value(42);
+}
```

Comment:
Are we testing the rust language now?
derives never modify the original type, so this test is useles.